api/v1beta2/moduletemplate_types.go

```diff
@@ -148,6 +154,14 @@ type ModuleTemplateList struct {
 	Items              []ModuleTemplate `json:"items"`
 }
 
+type Resource struct {
+	// Name is the name of the resource.
+	Name string `json:"name"`
```

Comment:
I was wondering if we want to somehow validate that if the resources array is not empty, then there should be an entry which has the name of "rawManifest"? I am not aware of any built in Kubebuilder markers that can validate this, but in general we do need to have this validation or?

Reply:
I think there would be "min items" (at least in JSON schema). But I was actually thinking about the same... The rawManifest is only necessary if the module supports manual installation which is not the case for all. But what we could do is to raise a warning in modulectl that "when missing rawManfiest in resources, the module template does not support manual installation".

Reply:
Also added this as a dicussion point to the modulectl related ticket